### PR TITLE
Feature/nft/flow approve allowance

### LINKED ISSUE #757: Merge NFT with royalty fees (CryptoAppoveAllowence)

### PR DESCRIPTION
**Pull request by rokn**
**Detailed description**:
This adds the new NFT flows which are based on Approve Allowance. Check #650 #651 for more info.

**New Things**:
As this implementation was already done earlier this year, I'll add just some things which I changed:

- Fixed a couple of bugs related to what is saved in the DB from the validators when the scheduled transaction is ApproveAllowance transaction. It is now correctly always adding the transfer with the new operation 'approve'.
- Fixed a couple of problems with the E2E tests, including:
    - Not waiting after the signatures are received
    - Added waits for the DB where necessary

**Which issue(s) this PR fixes**:
Fixes #757 

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: app/clients/hedera/mirror-node/client.go [hunk 1]
```diff
@@ -434,6 +426,21 @@ func (c Client) TopicExists(topicID hedera.TopicID) bool {
 	return c.query(topicQuery, topicID.String())
 }
 
+func (c *Client) GetTransactionsAfterTimestamp(accountId hedera.AccountID, startTimestamp int64, transactionType string) ([]transaction.Transaction, error) {
```
Comment (svetlio8) [inline app/clients/hedera/mirror-node/client.go:429]: Could you please add tests for this new func?

File: app/clients/hedera/mirror-node/model/transaction/transaction.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ package transaction
 
 import (
 	"errors"
+
```
Comment (svetlio8) [inline app/clients/hedera/mirror-node/model/transaction/transaction.go:21]: nit: not needed

File: app/clients/hedera/mirror-node/model/transaction/transaction.go [hunk 3]
```diff
@@ -151,6 +152,17 @@ func (t Transaction) GetHBARTransfer(account string) (amount int64, isFound bool
 	return 0, false
 }
 
+// GetTokenTransfer gets the token transfer for an Account
+func (t Transaction) GetTokenTransfer(account string) (amount int64, isFound bool) {
```
Comment (svetlio8) [inline app/clients/hedera/mirror-node/model/transaction/transaction.go:156]: Could you please add tests for this new func?

File: app/helper/fee/fee.go [hunk 4]
```diff
@@ -39,3 +40,28 @@ func TotalFeeFromTransfers(transfers []model.Hedera, receiver hedera.AccountID)
 
 	return strconv.FormatInt(result, 10), hasReceiver
 }
+
+// SumFallbackFeeAmounts sums fallback fees in HBAR and by token ID
+// Returns the sum of the fallback fees in HBAR and by token ID
+func SumFallbackFeeAmounts(customFees asset.CustomFees) asset.CustomFeeTotalAmounts {
```
Comment (svetlio8) [inline app/helper/fee/fee.go:46]: Could you please add tests for this new func?

File: app/helper/fee/fee.go [hunk 5]
```diff
@@ -39,3 +40,28 @@ func TotalFeeFromTransfers(transfers []model.Hedera, receiver hedera.AccountID)
...
+	customFeeAmounts := new(asset.CustomFeeTotalAmounts)
+	customFeeAmounts.FallbackFeeAmountsByTokenId = make(map[string]int64)
+	sumFallbackFeeAmounts(customFees.RoyaltyFees, customFeeAmounts)
+
+	return *customFeeAmounts
+}
+
+func sumFallbackFeeAmounts(royaltyFees []asset.RoyaltyFee, customFeeAmounts *asset.CustomFeeTotalAmounts) {
```
Comment (svetlio8) [inline app/helper/fee/fee.go:54]: Could you please add tests for this new func?

**Review by svetlio8 — APPROVED** [review]

**Review by trayanm — DISMISSED** [review]

**Review by svetlio8 — DISMISSED** [review]

**Review by Coiling-Dragon — APPROVED** [review]

**Review by svetlio8 — APPROVED** [review]

**Pull request merged**